external-providers/java-external-provider/pkg/java_external_provider/dependency.go

```diff
@@ -51,10 +67,41 @@ func (p *javaServiceClient) findPom() string {
 	if err != nil {
 		return ""
 	}
+	if _, err := os.Stat(f); errors.Is(err, os.ErrNotExist) {
+		return ""
+	}
 	return f
 }
 
+func (p *javaServiceClient) findGradleBuild() string {
+	// TODO: naive?
+	if p.config.Location != "" {
+		f, err := filepath.Abs(filepath.Join(p.config.Location, "build.gradle"))
+		if err != nil {
+			return ""
+		}
+		return f
+	}
+	return ""
+}
+
 func (p *javaServiceClient) GetDependencies(ctx context.Context) (map[uri.URI][]*provider.Dep, error) {
+	if p.getBuildTool() == gradle {
```

Comment:
is this redundant given that there's a switch in GetDepenenciesDAG()? I think we will need to move fallback call inside GetDepenciesDAG()